pkg/registry/rbac/reconciliation/reconcile_role.go

```diff
@@ -195,6 +202,20 @@ func computeReconciledRole(existing, expected RuleOwner, removeExtraPermissions
 		result.Operation = ReconcileUpdate
 	}
 
+	switch {
+	case existing.GetAggregationRule() == nil && expected.GetAggregationRule() == nil:
+	case existing.GetAggregationRule() != nil && expected.GetAggregationRule() == nil:
+		result.ReplacementAggregationRule = nil
+		result.Role.SetSetAggregationRule(nil)
+		result.Operation = ReconcileUpdate
+	case existing.GetAggregationRule() == nil && expected.GetAggregationRule() != nil,
+		// it is uncommon to have aggregation rules, just always reconcile these
+		existing.GetAggregationRule() != nil && expected.GetAggregationRule() != nil:
```

Comment:
when both are non-nil, I'm wary of stomping what was already there

Reply:
> when both are non-nil, I'm wary of stomping what was already there

I did make it a slice.  Would you like to combine them?

Reply:
I think I'd expect the default ones to be added if not covered by one already in the slice. auto-reconciliation is strictly additive today